packages/compiler-cli/src/ngtsc/core/src/compiler.ts

```diff
@@ -656,6 +657,26 @@ export class NgCompiler {
     return generateAnalysis(context);
   }
 
+  /**
+   * Gets information for the current program that may be used to generate API
+   * reference documentation. This includes Angular-specific information, such
+   * as component inputs and outputs.
+   */
+  getApiDocumentation(): DocEntry[] {
+    const compilation = this.ensureAnalyzed();
+    const checker = this.inputProgram.getTypeChecker();
+    const docsExtractor = new DocsExtractor(checker, compilation.metaReader);
+
+    let entries: DocEntry[] = [];
+    for (const sourceFile of this.inputProgram.getSourceFiles()) {
+      // We don't want to generate docs for `.d.ts` files.
+      if (sourceFile.isDeclarationFile) continue;
+
+      entries = entries.concat(docsExtractor.extract(sourceFile));
+    }
+    return entries;
```

Comment:
```suggestion
    const entries: DocEntry[] = [];
    for (const sourceFile of this.inputProgram.getSourceFiles()) {
      // We don't want to generate docs for `.d.ts` files.
      if (sourceFile.isDeclarationFile) continue;

      entries.push(...docsExtractor.extract(sourceFile));
    }
    return entries;
```



I'm not sure if there's a specific style in the compiler code for `concat` vs `push`, but certainly `push` is used a lot   more. I'm feeling like this should just be `push` instead.

Reply:
I had mentioned in one of Paul's comments; I used `concat` because my understanding was that it's usually faster than destructuring. Changed it, though, since I don't feel strongly enough and any given source file is unlikely to have more than a few stop-level statements anyway.